peter_rockwood/HW1.md

```diff
@@ -0,0 +1,234 @@
+
+### Homework #1
+(Due Monday 1/12)
+
+You need to turn in this homework by cloning this repo, making a new branch with your solutions, pushing to github, and issuing a pull request.
+For details, follow the instructions [here](http://portlandcodeschool.github.io/jse/2015/01/07/command-line-and-git-slides/#/14).
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** (_Difficulty: mixed_)
+Predict the output of each of the following expressions, and explain the rules which determine it.
+If the output depends on the value of x, identify the conditions when the expression will output true.  Assume the cases are independent, and x is reset to an unknown value before each.
+
+Some of these are tricky!  Don't trust your first instinct.  
+
+
+**a)** `"1" == 1`
+
+	True, type-converting equality.     					
+
+**b)** `"1" === 1`
+
+	False, strict equality.    					
+
+**c)** `x == 'x'`  
+
+	True if the value of x is 'x', otherwise false.   					
+
+**d)** `x == (x+'')`  
+
+	True, the empty string is ignored.					
```

Comment:
There are three cases where this isn't true: with `NaN`, because it's not equal to anything, including itself; and with either `true` or `false`, because their values will be autoconverted to strings when added to strings.